### PR TITLE
add state field in item and update related code

### PR DESCRIPTION
**Pull request by hewonjeong**
- add state filed in item
- update mockdata : add state field
- update remove endpoint in item API
- update add endpoint in item API
- update changeState function to change item's state field
- modify test Items key in getByArea test code.
   because it was occurred error because it had invalid key.
- Change vars name in test code. status -> state

### CODE REVIEW

File: src/server/database-mock-data.js [hunk 1]
```diff
@@ -89,6 +95,7 @@ export const constexpiredItem = {
   category: 'warning',
```
Comment (ovekyc) [inline src/server/database-mock-data.js]: You can import constant from key-utils

File: src/server/database-mock-data.js [hunk 2]
```diff
@@ -89,6 +95,7 @@ export const constexpiredItem = {
...
   startTime: '2016-10-10T01:11:46.851Z',
   endTime: '1999-01-01T10:10:10.851Z',
+  state: 'alive',
```
Comment (ovekyc) [inline src/server/database-mock-data.js]: ditto

File: src/server/database-mock-data.js (hunk 1 above)
Comment (ovekyc) [reply src/server/database-mock-data.js]: But it is not necessary now

File: src/server/items/models.js [hunk 3]
```diff
@@ -1,6 +1,11 @@
 import db from '../database';
 import {KeyUtils, STATE, CATEGORY, DEFAULT_PRECISON} from '../key-utils';
 
+export const STATE_STRING = {
```
Comment (ovekyc) [inline src/server/items/models.js:4]: Is it proper that locate this constant at key-utils?
And import it?

File: src/test/node/items/models.js [hunk 4]
```diff
@@ -14,8 +14,9 @@ const testItem = {
   category: 'event',
```
Comment (ovekyc) [inline src/test/node/items/models.js]: You can also import CATEGORY constant

File: src/test/node/users/models.js [hunk 5]
```diff
@@ -13,6 +14,7 @@ export const mockItem = {
   category: 'warning',
```
Comment (ovekyc) [inline src/test/node/users/models.js]: ditto

File: src/server/items/models.js (hunk 3 above)
Comment (hewonjeong) [reply src/server/items/models.js:4]: I thought this var have to be here cause it's not about 'key'.
Anyway i think so it's ambiguous. How you think which one is more proper?

File: src/test/node/items/controllers.js [hunk 6]
```diff
@@ -369,29 +372,34 @@ test('delete an item from database', t => {
...
-        // in case of index keys
-        const statusCode = data.key.charAt(STATE_CODE_POS);
-        t.equal(statusCode, expected.statusCode, 'should be same statusCode');
-      } else {
-        // in case of an original key
-        t.equal(data.value.title, expected.title);
-      }
-    }).on('error', (err) => {
-      t.fail('Error while reading from DB');
+    }).then(()=>{
```
Comment (ovekyc) [inline src/test/node/items/controllers.js:375]: It looks good.
But you also can use fetchPrefix function with array.map

**Comment by ovekyc** [comment]
LGTM

File: src/test/node/items/controllers.js (hunk 6 above)
Comment (hewonjeong) [reply src/test/node/items/controllers.js:375]: good. I can't sure that using fetchPrefix way is more better in performance,
but It improves readability and comfortability to write code.  I will try with that way,

**Comment by festinalente91** [comment]
LGTM

**Comment by hewonjeong** [comment]
LGTM

**Pull request merged**